tests/gem5/stats-reset/configs/x86-ubuntu-run-base.py

```diff
@@ -0,0 +1,125 @@
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+This script is based on
+configs/example/gem5_library/x86-ubuntu-run-with-kvm-no-perf.py
+
+This simulation runs for 15 billion ticks, then dumps the stats and exits.
+By itself, it doesn't test if the stats reset correctly. However, the stats.txt
+file it generates is used as a baseline comparison for the other two tests.
+
+It is only used to generate a reference file, so it is not run as part of the tests.
+"""
+
+import shutil
+from pathlib import Path
+
+import m5
+from m5.stats.gem5stats import get_simstat
+
+from gem5.components.boards.x86_board import X86Board
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.components.processors.simple_switchable_processor import (
+    SimpleSwitchableProcessor,
+)
+from gem5.isas import ISA
+from gem5.resources.resource import obtain_resource
+from gem5.simulate.exit_event import ExitEvent
+from gem5.simulate.simulator import Simulator
+from gem5.utils.requires import requires
+
+requires(
+    isa_required=ISA.X86,
+)
+
+from gem5.components.cachehierarchies.classic.private_l1_shared_l2_cache_hierarchy import (
+    PrivateL1SharedL2CacheHierarchy,
+)
+
+cache_hierarchy = PrivateL1SharedL2CacheHierarchy(
+    l1d_size="64KiB", l1i_size="64KiB", l2_size="1MiB"
+)
+
+from gem5.components.memory import DualChannelDDR4_2400
+
+memory = DualChannelDDR4_2400(size="3GiB")
+
+processor = SimpleSwitchableProcessor(
+    starting_core_type=CPUTypes.ATOMIC,
+    switch_core_type=CPUTypes.TIMING,
+    isa=ISA.X86,
+    num_cores=2,
+)
+
+board = X86Board(
+    clk_freq="3GHz",
+    processor=processor,
+    memory=memory,
+    cache_hierarchy=cache_hierarchy,
+)
+
+workload = obtain_resource("x86-ubuntu-24.04-boot-with-systemd")
+board.set_workload(workload)
+
+
+def exit_event_handler():
+    print("First exit: kernel booted")
+    processor.switch()
+    yield False  # gem5 is now executing systemd startup
+    print("Second exit: Started `after_boot.sh` script")
+    # m5.stats.dump()
+    yield False  # gem5 is now executing the `after_boot.sh` script
+    print("Third exit: Finished `after_boot.sh` script")
+    # The after_boot.sh script will run a script if it is passed via
+    # m5 readfile. This is the last exit event before the simulation exits.
+    yield True
+
+
+def max_tick_handler_no_reset():
+    m5.stats.dump()
+    # Move the stats file to the tests directory so it can be accessed by the
+    # next two tests
+    shutil.move(
+        f"{m5.options.outdir}/stats.txt",
+        "./gem5/stats-reset/base-case-stats.txt",
+    )
+    yield True
+
+
+simulator = Simulator(
+    board=board,
+    on_exit_event={
+        # Here we want override the default behavior for the first m5 exit
+        # exit event.
+        ExitEvent.EXIT: exit_event_handler(),
+        ExitEvent.MAX_TICK: max_tick_handler_no_reset(),
+    },
+)
+
+simulator.set_max_ticks(15000000000)  # 15,000,000,000
```

Comment:
FYI, You can use underscores in python to make big numbers more readable.
```suggestion
simulator.set_max_ticks(15_000_000_000)  # 15 ms
```